gm4_vigere_shamir/data/gm4_vigere_shamir/functions/heal.mcfunction

```diff
@@ -0,0 +1,44 @@
+
+# lower player's max health to their current health
+execute if score remove_health gm4_vs_data matches 512.. run attribute @s minecraft:generic.max_health modifier add e0c18b32-4caf-4193-8899-74ad1cc0dba2 gm4_vs_remove_512 -512 add
+execute if score remove_health gm4_vs_data matches 512.. run scoreboard players remove remove_health gm4_vs_data 512
+execute if score remove_health gm4_vs_data matches 256.. run attribute @s minecraft:generic.max_health modifier add f7676355-5de1-4167-9c01-2a274b5ee6bc gm4_vs_remove_256 -256 add
+execute if score remove_health gm4_vs_data matches 256.. run scoreboard players remove remove_health gm4_vs_data 256
+execute if score remove_health gm4_vs_data matches 128.. run attribute @s minecraft:generic.max_health modifier add 2df88b31-a47f-4c31-a5b8-aa51ed1c35bb gm4_vs_remove_128 -128 add
+execute if score remove_health gm4_vs_data matches 128.. run scoreboard players remove remove_health gm4_vs_data 128
+execute if score remove_health gm4_vs_data matches 64.. run attribute @s minecraft:generic.max_health modifier add 41fd129e-322f-4147-b2fa-8185206f5b8e gm4_vs_remove_64 -64 add
+execute if score remove_health gm4_vs_data matches 64.. run scoreboard players remove remove_health gm4_vs_data 64
+execute if score remove_health gm4_vs_data matches 32.. run attribute @s minecraft:generic.max_health modifier add 1afb3106-b947-46ca-942a-cad1a78f5df7 gm4_vs_remove_32 -32 add
+execute if score remove_health gm4_vs_data matches 32.. run scoreboard players remove remove_health gm4_vs_data 32
+execute if score remove_health gm4_vs_data matches 16.. run attribute @s minecraft:generic.max_health modifier add 04f2b196-ca85-4ac8-b8e0-a7d569b415c8 gm4_vs_remove_16 -16 add
+execute if score remove_health gm4_vs_data matches 16.. run scoreboard players remove remove_health gm4_vs_data 16
+execute if score remove_health gm4_vs_data matches 8.. run attribute @s minecraft:generic.max_health modifier add 1f624b8a-89a4-4951-b4ce-f21890adbf89 gm4_vs_remove_8 -8 add
+execute if score remove_health gm4_vs_data matches 8.. run scoreboard players remove remove_health gm4_vs_data 8
+execute if score remove_health gm4_vs_data matches 4.. run attribute @s minecraft:generic.max_health modifier add 52b90103-a8c1-40a1-b2dd-9e7feb978b52 gm4_vs_remove_4 -4 add
+execute if score remove_health gm4_vs_data matches 4.. run scoreboard players remove remove_health gm4_vs_data 4
+execute if score remove_health gm4_vs_data matches 2.. run attribute @s minecraft:generic.max_health modifier add c875bec0-f7e4-4479-a030-dc92eb39d29a gm4_vs_remove_2 -2 add
+execute if score remove_health gm4_vs_data matches 2.. run scoreboard players remove remove_health gm4_vs_data 2
+execute if score remove_health gm4_vs_data matches 1.. run attribute @s minecraft:generic.max_health modifier add 41b9fe4f-ce8e-4025-8215-d47f9321b853 gm4_vs_remove_1 -1 add
+execute if score remove_health gm4_vs_data matches 1.. run scoreboard players remove remove_health gm4_vs_data 1
```

Comment:
You could reuse the objective `gm4_ml_data`, as metallurgy will always be installed alongside this.